docs/agda-spec/src/Spec/Protocol.lagda

```diff
@@ -0,0 +1,172 @@
+\subsection{$\mathsf{PRTCL}$ Transition}
+\label{sec:protocol-trans}
+
+The Protocol Transition ($\mathsf{PRTCL}$) calls the transition $\mathsf{UPDN}$ to update the evolving
+and candidate nonces, and checks the operational certificate with $\mathsf{OCERT}$. Its environment is shown in
+Figure~\ref{fig:ts-types:prtcl} and consists of:
+
+\begin{itemize}
+  \item The stake pool stake distribution \afld{pd}.
+  \item The epoch nonce \afld{η₀}.
+\end{itemize}
+
+Its state is shown in Figure~\ref{fig:ts-types:prtcl} and consists of
+\begin{itemize}
+  \item The operational certificate issue number mapping \afld{cs}.
+  \item The evolving nonce \afld{ηv}.
+  \item The candidate nonce for the next epoch \afld{ηc}.
+\end{itemize}
+
+\begin{code}[hide]
+{-# OPTIONS --safe #-}
+
+open import Spec.BaseTypes using (Nonces)
+open import Spec.BlockDefinitions
+open import Ledger.Crypto
+open import Ledger.Script
+open import Ledger.Types.Epoch
+open import Data.Rational.Ext
+
+module Spec.Protocol
+  (crypto : _) (open Crypto crypto)
+  (nonces : Nonces crypto) (open Nonces nonces)
+  (es     : _) (open EpochStructure es)
+  (ss     : ScriptStructure crypto es) (open ScriptStructure ss)  
+  (bs     : BlockStructure crypto nonces es ss) (open BlockStructure bs)
+  (af     : _) (open AbstractFunctions af)
+  (rs     : _) (open RationalExtStructure rs)
+  where
+
+open import InterfaceLibrary.Common.BaseTypes crypto using (PoolDistr; lookupPoolDistr)
+open import Spec.OperationalCertificate crypto nonces es ss bs af
+open import Spec.UpdateNonce crypto nonces es
+open import Spec.BaseTypes crypto using (OCertCounters)
+open import Data.Rational as ℚ using (ℚ; 0ℚ; 1ℚ)
+open import Ledger.Prelude
+open Ledger.Prelude.ℤ using (pos)
+
+\end{code}
+
+\begin{figure*}[h]
+\emph{Protocol environments}
+\begin{code}
+record PrtclEnv : Type where
+  constructor ⟦_,_⟧ᵖᵉ
+  field pd  : PoolDistr -- pool stake distribution
+        η₀  : Nonce     -- epoch nonce
+\end{code}
+\emph{Protocol states}
+\begin{code}
+record PrtclState : Type where
+  constructor ⟦_,_,_⟧ᵖˢ
+  field cs : OCertCounters -- operational certificate issues numbers
+        ηv : Nonce         -- evolving nonce
+        ηc : Nonce         -- candidate nonce
+\end{code}
+\emph{Protocol transitions}
+\begin{code}[hide]
+data
+\end{code}
+\begin{code}
+  _⊢_⇀⦇_,PRTCL⦈_ : PrtclEnv → PrtclState → BHeader → PrtclState → Type
+\end{code}
+\caption{Protocol transition system types}
+\label{fig:ts-types:prtcl}
+\end{figure*}
+
+\begin{figure*}[h]
+\emph{Protocol helper functions}
+\begin{code}
+hBLeader : BHBody → ∃[ n ] n < 2 ^ 512 -- [0, 2^512)

```

Comment:
I think it'd make sense to define an alias for this type. It comes up a bunch and people that don't know dependent types will be confused.

Reply:
Fixed in 8a72ee453fe1f069bd778b95895e9b0d6b9d54ef.